File: python/167. Two Sum II - Input Array Is Sorted.py
from typing import List

class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        ptr2: int = len(numbers) - 1
        ptr1: int = 0
        while(ptr1 < ptr2):
            if(numbers[ptr1] + numbers[ptr2] == target):
                return [ptr1+1, ptr2+1]
            elif(numbers[ptr1] + numbers[ptr2] < target):
                ptr1+=1
            else:
                ptr2-=1    

solution = Solution()

#Input: numbers = [2,7,11,15], target = 9
#Output: [1,2]
print(solution.twoSum([2,7,11,15], 9))    


#Input: numbers = [2,3,4], target = 6
#Output: [1,3]
print(solution.twoSum([2,3,4], 6))   

File: java/167. Two Sum II - Input Array Is Sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int ptr1 = 0, ptr2 = numbers.length - 1;
        int[] result = new int[2];
        while(ptr1 < ptr2) {
            if(numbers[ptr1] + numbers[ptr2] == target) {
                result[0] = ptr1+1;
                result[1] = ptr2+1;
                return result;
            } else if(numbers[ptr1] + numbers[ptr2] < target) 
                ptr1+=1;
            else
                ptr2-=1;       
        }
        return result;
    }

    public static void main(String args[]) {
        Solution solution = new Solution();
        System.out.println(List.of(solution.twoSum(new int[]{2,7,11,15}, 9)))
    }
}